### ISSUE TITLE
[g2o] cmake config file missing transitive dependency 

### ISSUE THREAD

**Issue by autoantwort** [microsoft/vcpkg#33041]
**Describe the bug**
The cmake config file of g2o only specifies `libmetis` as `INTERFACE_LINK_LIBRARIES` in `share/g2o/g2oTargets.cmake` after installation.  
But mentis specifies `INTERFACE_LINK_LIBRARIES "GKlib"` in `share/metis/metisTargets.cmake` after installation. 

**To Reproduce**
1. ./vcpkg install g2o
2. See error

**Expected behavior**
The cmake config file of g2o should mention the `GKlib` lib dependency or link against the metis target.

**Comment by MonicaLiu0311** [comment]
> To Reproduce
> Steps to reproduce the behavior:
> 
> 1. ./vcpkg install g2o
> 2. See error

* x64-windows
* vcpkg version:
  ```
  PS F:\vcpkg> ./vcpkg version
  vcpkg package management program version 2023-08-02-6d13efa755f9b5e101712d210199e4139b4c29f6

  See LICENSE.txt for license information.
  ```

Sorry I didn't repro your issue, maybe I'm misunderstanding something?

```
PS F:\vcpkg> ./vcpkg install g2o:x64-windows
Computing installation plan...
The following packages will be built and installed:
    g2o:x64-windows -> 2020-02-07#4
Detecting compiler hash for triplet x64-windows...
Restored 1 package(s) from C:\Users\v-liumonica\AppData\Local\vcpkg\archives in 6 s. Use --debug to see more details.
Installing 1/1 g2o:x64-windows...
Elapsed time to handle g2o:x64-windows: 2.5 s
Total install time: 2.5 s
g2o provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(g2o CONFIG REQUIRED)
    # note: 17 additional targets are not displayed.
    target_link_libraries(main PRIVATE g2o::core g2o::stuff g2o::types_icp g2o::types_sba)
```

**Comment by autoantwort** [comment]
@MonicaLiu0311 Please remove the https://github.com/microsoft/vcpkg/labels/requires%3Arepro label

**Comment by MonicaLiu0311** [comment]
Fixed by #42600.
